src/App.tsx

```diff
@@ -1,17 +1,40 @@
+import { useEffect, useState } from 'react';
 import { Routes, Route } from 'react-router-dom';
 import Navigation from '@/features/Navigation';
 import Footer from '@/components/Footer';
 import MentorPage from '@/features/MentorPage';
+import Logout from '@/features/Logout';
 
 function App() {
+  const [authenticated, setAuthenticated] = useState(false);
+  useEffect(() => {
+    const authenticateUser = async () => {
+      const response = await fetch('/api/myuser');
+      if (response.redirected) {
+        // If there is no active session, the user is redirected to the login page
+        window.location.href = response.url;
+      } else if (response.ok) {
+        setAuthenticated(true);
+      } else {
+        window.location.href = '/login';
+      }
+    };
+    authenticateUser();
+  }, []);
```

Comment:
Could we extract this piece of code to a hook like useAuthenticated?
Like this:
![image](https://user-images.githubusercontent.com/34128180/206563161-233a0267-012d-449d-b78e-fbb0699ed72b.png)

Reply:
Ah yes, I'll do it!

Reply:
The hook is now extracted!